### PR TITLE
Add missing breadcrumb links in order pages: confirmation/return/follow, cart and addresses pages

### LINKED ISSUE #17131: [EPIC] - Standardize breadcrumbs in Front-office
# Breadcrumbs in the Front-office

## Goal
The goal of this EPIC is to clarify the rule about breadcrumbs in PrestaShop Front-office and to adapt existing/add missing breadcrumbs according to it.

## Why
Several issues have been opened in the previous months about breadcrumbs. 
Some PRs on this topic have been submitted as well.

As of today, there is no clear rule defined about breadcrumbs in the FO. We need to define one and to implement it.

## Related issues

- [x] #29539

### Related issues
- [x] #15440 - No breadcrumb on some important pages (like Stores, Sitemap, New Products, Best Sales, Suppliers, Cart...)
- [x] #16836 - Breadcrumb missing on my account page
- [x] #22416 - SEO-Problem: Breadcrumblist error
- [x] #23530 - Desactivated category still display in breadcrumbs
- [x] #26261

### Specs
- [Specs for breadcrumbs in Front-office ](https://github.com/PrestaShop/prestashop-specs/pull/88/files )

### Implementation 
Missing breadcrumbs to add:
- [ ] TBC

Existing breadcrumbs to remove:
- [ ] Breadcrumbs from CMS pages because they don't have more than 2 depths. 

Existing breadcrumbs to remove when user isn't logged in:
- [ ] /index.php?controller=Identity
- [ ] /index.php?controller=OrderHistory
- [ ] /index.php?controller=CreditSlip
- [ ] /index.php?controller=Addresses

Existing breadcrumbs to modify:
- [ ] /index.php?controller=Search (product page): 
Replace 'Home > Clothes > Women > Hummingbird t-shirt' 
by Home > Search results for hummingbird > Hummingbird printed t-shirt'

### PR DESCRIPTION
**Pull request by nprokopenko202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add breadcrumb links to pages: cart, order confirmation, customer merchandise returns, return details, 4th breadcrumb level for addresses: update address, create new address
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17131.
| How to test?  | 

Go to following pages to reproduce:
- /index.php?controller=OrderFollow > added 'Merchandise returns' link
- /index.php?controller=OrderReturn > : added Return number as 4th level of breadcrumb
- /index.php?controller=OrderConfirmation > : added 'Order confirmation' link
- /index.php?controller=Address > : added 'Update your address' or 'New address' as 4th level

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19298)

### CODE REVIEW

File: controllers/front/AddressController.php [hunk 1]
```diff
@@ -138,6 +138,18 @@ public function getBreadcrumbLinks()
             'url' => $this->context->link->getPageLink('addresses'),
         ];
 
+        $title = $this->trans('New address', [], 'Shop.Theme.Customeraccount');
+
+        $id_address = Tools::getValue('id_address');
+        if ($id_address) {
+            $title = $this->trans('Update your address', [], 'Shop.Theme.Customeraccount');
+        }
```
Comment (Progi1984) [inline controllers/front/AddressController.php]:
```suggestion
        $id_address = Tools::getValue('id_address');
        $title = $id_address
            ? $this->trans('Update your address', [], 'Shop.Theme.Customeraccount')
            : $this->trans('New address', [], 'Shop.Theme.Customeraccount');
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback @nprokopenko202 ;)

**Review by Progi1984 — DISMISSED** [review]

**Comment by LouiseBonnard** [comment]
Hi @Progi1984, you can request @Junebyun's review as well. :-)

**Review by LouiseBonnard — DISMISSED** [review]

**Review by matks — DISMISSED** [review]
Nice work !

**Comment by matks** [comment]
When this PR is merged, we will be able to close the EPIC 🎉

**Review by PierreRambaud — DISMISSED** [review]

**Comment by nprokopenko202** [comment]
Hello everyone,
I've removed breadcrumb from Cart page as mentioned in this comment:
https://github.com/PrestaShop/PrestaShop/issues/17131#issuecomment-633973331

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @nprokopenko202  ! 

PS : I have edited the description of this PR just to remove this 

> * /index.php?controller=Cart > added 'Cart' link

**Comment by Progi1984** [comment]
Thanks @nprokopenko202

**Pull request merged**